packages/synthetix-main/contracts/modules/core/PoolModule.sol

```diff
@@ -136,26 +131,26 @@ contract PoolModule is IPoolModule {
                 }
                 lastMarketId = markets[i];
 
-                MarketDistribution.Data memory distribution;
-                distribution.market = markets[i];
-                distribution.weight = uint128(weights[i]);
-                distribution.maxDebtShareValue = int128(maxDebtShareValues[i]);
+                MarketConfiguration.Data memory marketConfig;
+                marketConfig.market = markets[i];
+                marketConfig.weight = uint128(weights[i]);
+                marketConfig.maxDebtShareValue = int128(maxDebtShareValues[i]);
 
-                pool.poolDistribution.push(distribution);
+                pool.marketConfigurations.push(marketConfig);
 
                 totalWeight += weights[i];
             }
         }
 
-        uint popped = pool.poolDistribution.length - i;
+        uint popped = pool.marketConfigurations.length - i;
         for (i = 0; i < popped; i++) {
-            Market.rebalance(pool.poolDistribution[pool.poolDistribution.length - 1].market, poolId, 0, 0);
-            pool.poolDistribution.pop();
+            Market.rebalance(pool.marketConfigurations[pool.marketConfigurations.length - 1].market, poolId, 0, 0);
+            pool.marketConfigurations.pop();
         }
 
         pool.totalWeights = uint128(totalWeight);
 
-        pool.rebalanceConfigurations();
```

Comment:
I found that `rebalanceConfigurations` only calls `distributeDebt`. Its an unnecessary wrapper, so I removed it.